public/app/features/alerting/unified/api/onCallApi.ts

```diff
@@ -8,13 +12,30 @@ export type OnCallIntegrationsUrls = string[];
 export const onCallApi = alertingApi.injectEndpoints({
   endpoints: (build) => ({
     getOnCallIntegrations: build.query<OnCallIntegrationsUrls, void>({
-      query: () => ({
-        headers: {},
-        url: '/api/plugin-proxy/grafana-oncall-app/api/internal/v1/alert_receive_channels/',
-      }),
+      queryFn: async () => {
+        try {
+          const integrations = await fetchOnCallIntegrations();
+          return { data: integrations };
+        } catch (error) {
+          return { error: error };
```

Comment:
Actually is this catch necessary at all? Since there's already a `catch` in `fetchOnCallIntegrations` which returns an empty array if it fails, so I wonder if this code will ever be executed.

Reply:
💯